lib/index.js

```diff
@@ -86,22 +87,27 @@ Optimizely.prototype.initialize = function() {
  */
 
 Optimizely.prototype.track = function(track) {
+  var opts = this.options;
+  var eventProperties = track.properties();
   // Use the new-style API (which is compatible with Classic and X)
   var payload = {
     type: 'event',
     eventName: track.event(),
-    tags: track.properties()
+    tags: eventProperties
   };
 
-  if (payload.tags.revenue) {
+  if (eventProperties.revenue && opts.sendRevenueOnlyForOrderCompleted) {
+    if (track.event() === 'Order Completed') {
+      payload.tags.revenue = Math.round(payload.tags.revenue * 100);
```

Comment:
Sorry, I do not seem to be able to describe the issue fully it seems. :( 
Please refer to the 2nd location where eventProperties is passed to the full stack. 
https://github.com/segment-integrations/analytics.js-integration-optimizely/blob/master/lib/index.js#L109
Full stack also needs revenue to be formatted with Math.round(payload.tags.revenue * 100);
This is why I suggested applying the math to eventProperties.revenue.